Automattic/woocommerce-payments
# Default WCPay gateway should always be present on WC Settings screen for Payments

Fixes #6279 

#### Changes proposed in this Pull Request


# Default WCPay gateway should always be present on WC Settings screen for Payments


The WCPay payment gateway supports multiple different methods and when the split-UPE is enabled these methods appear as separate payment gateways. [We manually remove these additional payment methods/gateways from the general WC Settings screen for Payments](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L701), to ensure that only the main WCPay gateway is present. 

When the split-UPE gateway is enabled, currently [we remove all but the legacy card gateway](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L701) from this list on the settings screen. However, this is incorrect, because when WooPay is disabled, the UPE card gateway is the main WCPay gateway instead. Consequently, when WooPay is disabled, this section of code ruthlessly removes all WCPay gateways, leading to the situation described in #6279.

This PR amends this section of code to maintain the correct gateway in all scenarios.  




#### Testing instructions


To get to the main WC Settings screen, visit **Payments > Settings** and select "All payment methods". WooCommerce Payments should always be present as a payment method on this screen. This should be true in all of the following situations:
- All UPE disabled
- Legacy UPE enabled
- Split-UPE enabled
   -  WooPay enabled
   - WooPay disabled


-------------------


**Post merge**
